cos-fleetshard-operator-camel/src/main/java/org/bf2/cos/fleetshard/operator/camel/CamelOperandSupport.java

```diff
@@ -373,25 +373,19 @@ public static Optional<KameletBinding> lookupBinding(KubernetesClient client, Ma
     }
 
     public static void computeStatus(ConnectorStatusSpec statusSpec, KameletBindingStatus kameletBindingStatus) {
-        if (kameletBindingStatus.phase != null) {
-            switch (kameletBindingStatus.phase.toLowerCase(Locale.US)) {
-                case KameletBindingStatus.PHASE_READY:
-                    statusSpec.setPhase(ManagedConnector.STATE_PROVISIONING);
-                    if (kameletBindingStatus.conditions != null) {
-                        boolean readyCondition = kameletBindingStatus.conditions.stream()
-                            .anyMatch(cond -> "Ready".equals(cond.getType())
-                                && "True".equals(cond.getStatus()));
-                        if (readyCondition) {
-                            statusSpec.setPhase(ManagedConnector.STATE_READY);
-                        }
+        statusSpec.setPhase(ManagedConnector.STATE_PROVISIONING);
+        if (kameletBindingStatus.conditions != null) {
+            Optional<Condition> readyConditionOpt = kameletBindingStatus.conditions.stream().filter(cond -> "Ready".equals(cond.getType())).findAny();
+            if (readyConditionOpt.isPresent()) {
+                Condition readyCondition = readyConditionOpt.get();
+                boolean klbReady =  "True".equals(readyCondition.getStatus());
+                if (klbReady) {
+                    statusSpec.setPhase(ManagedConnector.STATE_READY);
+                } else {
+                    if(!readyCondition.getMessage().isEmpty()) {
```

Comment:
I'm not sure if this is 100% correct as I think - but it needs to be double checked - the ready condition may be `False` but because the deployment is in progress hence the only presence of a message is not an indication that there is a failure. I believe we must check the content of the message at this stage.

But @rinaldodev may know better

Reply:
With failure `KameletBindingStatus` looks like
```
status:
  conditions:
    - lastTransitionTime: '2023-03-30T09:33:12Z'
      lastUpdateTime: '2023-03-30T09:33:12Z'
      message: >-
        [Pod mctr-cgila9m8qbfm7ltin430-9f686bb74-2mml6 runtime is not ready:
        map[bootstrap.servers:vajain-rho-cgijrtpm-iq--tu-nbaa.bf2.kafka.rhcloud.com:443
        check.kind:READINESS client.id:consumer-cgila9m8qbfm7ltin430-1
        group.id:cgila9m8qbfm7ltin430 route.id:cgila9m8qbfm7ltin430-source
        topic:vajain_topic]]
      reason: RuntimeNotReady
      status: 'False'
      type: Ready
  observedGeneration: 1
  phase: Ready
  replicas: 1
  selector: camel.apache.org/integration=mctr-cgila9m8qbfm7ltin430
  ```

Reply:
I had done a quick test with the changes of this PR, its working fine for atleast this particular scenario. 
But I agree with @lburgazzoli concern, might be its not valid for all scenario. Best I could think is to run the all intgeration test and see if everything works as expected.

Reply:
I agree with Luca here. We need to make sure that having a message actually means a failure.

By looking at Camel K code, it looks we could do that if we filter out those 2 phases:

```
	// KameletBindingPhaseNone --
	KameletBindingPhaseNone KameletBindingPhase = ""
	// KameletBindingPhaseCreating --
	KameletBindingPhaseCreating KameletBindingPhase = "Creating"
```

During the initialization, when the KameletBinding is in one of those 2 phases, it will be "Ready=false". If at that same moment a message is present, that may not represent a failure, just a deployment that is in progress.

https://github.com/bf2fc6cc711aee1a0c2a/cos-fleetshard/pull/637/files#diff-8c06870edc5b0b037fa5742434a6a98dd9fd3fe3b90df639f9847382c3082239R386-R388

Reply:
@lucamolteni @rinaldodev 
While analysing Camel-K operator code, I notice that Camel-k has already fixed this issue in `v1.12.0` to mark Integration PHASE to `ERROR` if POD is not recoverable.
I also tested our application with Camel-K `v1.12.0`, this behaviour is wroking as expected and `cos-fleetshard-camelk-operator` has marked the `ManagedConnector` resource to `failed` state without any change of our code.
https://github.com/apache/camel-k/blob/main/pkg/controller/integration/monitor.go#L423

Could we upgrade Camel-K operator version to `v1.12.0`? or we still need to implement some hotfix in our application?

Reply:
I had push the changes to monitor Error message `runtime not ready`.

Reply:
I made some more changes in my PR
rather then depends on `message`, I put the check on `reason`

`message` values are derived from different objects such as pod Condition, Container Condition, Knative Condition, Integration Kit Condition etc. Its getting very hard to track what exact message each of these object sends
I found that Camel-K already have some logic to segregate these failure condition and then always set specific `Reason = Error` when situation is un-recoverable